aws/routeAwsGet.go

```diff
@@ -18,21 +18,73 @@ import (
 	"database/sql"
 	"errors"
 	"net/http"
+	"strconv"
+	"strings"
 
 	"github.com/trackit/jsonlog"
 
 	"github.com/trackit/trackit2/db"
+	"github.com/trackit/trackit2/models"
 	"github.com/trackit/trackit2/routes"
 	"github.com/trackit/trackit2/users"
 )
 
+// AwsAccountsFromUserIDByAccountID retrieves rows from 'trackit.aws_account' as AwsAccount.
+// The result is filtered by a slice of accountID
+func AwsAccountsFromUserIDByAccountID(db models.XODB, userID int, accountIDs []int) ([]AwsAccount, error) {
+	var err error
+	var stringAccountIDs []string
+
+	// gen account_id
+	for _, id := range accountIDs {
+		stringAccountIDs = append(stringAccountIDs, strconv.FormatInt(int64(id), 10))
+	}
+	accountID := "(" + strings.Join(stringAccountIDs, ",") + ")"
+
+	// sql query
+	var sqlstr = `SELECT ` +
+		`id, user_id, pretty, role_arn, external ` +
+		`FROM trackit.aws_account ` +
+		`WHERE user_id = ? ` +
+		`AND id IN ` + accountID
```

Comment:
We currently use MySQL and the driver does not support directly using slices directly here so this can’t get much better.